### DISCUSSION TITLE
Is invokeid attribute set in meta if a child machine triggers an effect?

### DISCUSSION THREAD

**Discussion by alexkuchun** [statelyai/xstate#4640] [Q&A] [answered]
Whenever we trigger an action effect in the parent machine from the child machine using `sendParent`, shouldn't the `meta` attribute for the parent action effect contain the `invokeid`?

In the example below, I am trying to access `invokeid` value, depending on which I update either one or another alert in the parent state. However, it ends up being `undefined`. According to the documentation though, it should always be set by the SCXML Processor if the event is generated from an invoked child process. So, are the child machines treated as such? Or am I  misunderstanding what 'generated from an invoked child process' means?

child machine

```
export type TriggerAlertEvent = {
    type: 'TRIGGER_ALERT'
    message?: string
}

type AlertMachineEvent = ... | TriggerAlertEvent | ...

export const alertMachine = createMachine<AlertMachineContext, AlertMachineEvent>({
    ...
    {
        actions: {
            ...
            onAlertResult: sendParent<
            AlertingMachineContext,
            AlertingMachineEvent,
            TriggerAlertEvent
            >(({ message }) => ({
                type: 'TRIGGER_ALERT',
                message,
            })),
        }
    }
})
```

parent machine

```
type ParentMachineContext = {
    messageAlertMachine1?: string
    messageAlertMachine2?: string
    ...
}

const machine = createMachine<ParentMachineContext, ParentMachineEvents>({
    id: 'parent-machine',
    initial: 'start',
    states:
        start: {
            type: 'parallel',
            states: {
                state1,
                state2
            }
        }
...
})

const state1 = {
     invoke: [
           {id: 'alertMachine1', src: alertMachine}
           {id: 'alertMachine2', src: alertMachine}
     ],
    on: {
        TRIGGER_ALERT: 
            actions: [
                assign<OurMachineContext, TriggerAlertEvent>({
                    messageAlertMachine1: (_, event, meta) => meta._event.invokeid === 'alertMachine1' ? event.message : undefined,
                    messageAlertMachine1: (_, event, meta) => meta._event.invokeid === 'alertMachine2' ? event.message : undefined,
                })
           ]
    }
...}
```

**Comment by Andarist** [comment, marked as answer]
IIRC we have never implemented support for `invokeid` but in v4 you might be able to utilize `meta._event.origin` for this.

I wouldn't recommend it though, it would be better to assign some unique data to each one of those (in v4 you can do this using `.withContext`) and pass this data explicitly back to the parent as part of the event.
